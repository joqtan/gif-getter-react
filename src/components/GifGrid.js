// import React, { useState, useEffect } from 'react'
import PropTypes from 'prop-types'
import useFetchGif from '../hooks/useFetchGif'
import GifGridItem from './GifGridItem'

const GifGrid = ({ category }) => {
    const { data, loading } = useFetchGif(category)

    return (
        <>
            <h3 className='animate__animated animate__jello'>{category}</h3>
            {loading && <p>loading data...</p>}
            <div className='card-grid'>
                {data.map((img) => (
                    <GifGridItem key={img.id} {...img} />
                ))}
            </div>
        </>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string,
}

export default GifGrid
