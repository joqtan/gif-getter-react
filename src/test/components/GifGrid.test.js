import { shallow } from 'enzyme'
import GifGrid from '../../components/GifGrid'
import useFetchGif from '../../hooks/useFetchGif'

jest.mock('../../hooks/useFetchGif')

describe('test on <GifGrid/> component', () => {
    const category = ''

    test('component should render properly', () => {
        useFetchGif.mockReturnValue({ data: [], loading: true })
        const wrapper = shallow(<GifGrid category={category} />)
        expect(wrapper).toMatchSnapshot()
    })

    test('should show items when images were loaded in useFetchGif', () => {
        const gifs = [
            {
                id: '72534333',
                url: 'https://www.giflink.com/img/id?72534333',
                title: 'GIF title',
            },
        ]
        useFetchGif.mockReturnValue({ data: gifs, loading: false })
        const wrapper = shallow(<GifGrid category={category} />)
        expect(wrapper.find('p').exists()).toBe(false)
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length)
    })
})
