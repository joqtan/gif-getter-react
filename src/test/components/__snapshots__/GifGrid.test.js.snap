// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test on <GifGrid/> component component should render properly 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__jello"
  />
  <p>
    loading data...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
